unitary/alpha/quantum_world.py

```diff
@@ -269,6 +290,52 @@ def undo_last_effect(self):
             raise IndexError("No effects to undo")
         self.circuit, self.post_selection = self.effect_history.pop()
 
+    def save_snapshot(self) -> None:
+        """Saves the current length of the effect history and qubit_remapping_dict."""
```

Comment:
I think we want a better description here.  Is the idea that we save a snapshot every move?